src/nvim/api/private/helpers.c

```diff
@@ -605,8 +606,13 @@ void modify_keymap(Buffer buffer, bool is_unmap, String mode, String lhs, String
     return;
   }
 
-  if (opts != NULL && opts->luaref.type == kObjectTypeLuaRef) {
-    lua_funcref = api_new_luaref(opts->luaref.data.luaref);
+  if (rhs_obj.type == kObjectTypeString) {
+    rhs = rhs_obj.data.string;
+  } else if (rhs_obj.type == kObjectTypeLuaRef) {
+    lua_funcref = api_new_luaref(rhs_obj.data.luaref);
+  } else {
+    api_set_error(err, kErrorTypeValidation, "Unexpected type of rhs provided");
+    return;
```

Comment:
I would recommend also adding a `desc` option for when `rhs` is a function (similar to #16752) that is listed when users use `:map {lhs}`.

Reply:
Good idea . I was initially thinking of adding it on a followup . But can be added right here .